### PR TITLE
Remove cloning

### PR DESCRIPTION
**Pull request by nils-ohlmeier**
Improved proper ownership and removed as many direct clone() calls as possible.

### CODE REVIEW

**Comment by nils-ohlmeier** [comment]
@na-g it's not fully at what we talked about it, but hopefully a step in the right direction. What do you think?

File: src/lib.rs [hunk 1]
```diff
@@ -211,6 +211,47 @@ impl SdpSession {
...
+                    });
+                }
+                SdpType::Media(_) => {
+                    return Err(SdpParserError::Sequence {
+                        message: "media line not allowed in session parser".to_string(),
+                        line_number: line.line_number,
+                    });
+                }
+                // the line parsers throw unsupported errors for these already
+                SdpType::Email(_)
```
Comment (na-g) [inline src/lib.rs]: If these SDP types throw parsing errors, then they should be removed from the enum, and that way we don't have to worry about illegal state everywhere we use this enum. This follows the '"make the impossible inexpressible" rule of thumb. That could be handled in another bug though.

File: src/lib.rs [hunk 2]
```diff
@@ -1011,7 +1026,7 @@ pub fn parse_sdp(sdp: &str, fail_on_warning: bool) -> Result<SdpSession, SdpPars
             line_number: 0,
         });
     }
-    if sdp.len() < 62 {
+    if sdp.len() < 40 {
         return Err(SdpParserError::Line {
             error: SdpParserInternalError::Generic("string to short to be valid SDP".to_string()),
```
Comment (na-g) [inline src/lib.rs]: "string to short" -> "string too short"

File: src/lib.rs (hunk 1 above)
Comment (nils-ohlmeier) [reply src/lib.rs]: Excellent idea. I created issue #107 for that!

File: src/lib.rs (hunk 2 above)
Comment (nils-ohlmeier) [reply src/lib.rs]: Thanks. Fixed.

**Comment by nils-ohlmeier** [comment]
> While it won't let me leave an inline comment that isn't close to one of the lines that you edited (grrr...), you can change line 1085 of lib.rs to `return Err(warnings.remove(0));`, and remove `#derive(Clone)` from `SdpParserError`, then LGTM.

Ha, awesome. Yes another clone() call gone. Thank you!

**Pull request merged**